src/types.ts

```diff
@@ -29,7 +44,70 @@ export type SlackManifestType = {
   outgoingDomains?: Array<string>;
   types?: ICustomType[];
   datastores?: ManifestDatastore[];
-};
+}
+
+/** ISlackManifestRemote contains the features currently available to
+ * ManifestSchema.settings.function_runtime = remote apps. Here we surface that value
+ * to the developer as the slackHosted property.
+ *
+ * A subset of properties overlap with ISlackManifestRemote
+ */
+export interface ISlackManifestRemote {
+  slackHosted: false; // maps to function_runtime = "remote" in ManifestSchema
+  name: string;
+  backgroundColor?: string;
+  description: string;
+  displayName?: string;
+  icon: string;
+  longDescription?: string;
+  botScopes: Array<string>;
+  functions?: ManifestFunction[];
+  workflows?: ManifestWorkflow[];
+
+  // Outgoing domains are not enforced for remote functions
+  // outgoingDomains?: Array<string>;
+  types?: ICustomType[];
+  datastores?: ManifestDatastore[];
+
+  // Features supported in Platform1_0 (Remote) apps only (remote / non-run on Slack)
+  settings?: Omit<
+    ManifestSettingsSchema,
+    | "function_runtime"
+    | "event_subscriptions"
+    | "socket_mode_enabled"
+    | "token_rotation_enabled"
+  >; // lifting omitted properties to top level
+  eventSubscriptions?: ManifestEventSubscriptionsSchema;
+  socketModeEnabled?: boolean;
+  tokenRotationEnabled?: boolean;
+
+  appDirectory?: ManifestAppDirectorySchema;
+
+  // oauth
+  userScopes?: Array<string>;
+  redirectUrls?: Array<string>;
+  tokenManagementEnabled?: boolean;
+
+  features?: ISlackManifestRemoteFeaturesSchema;
+}
+
+export interface ISlackManifestRemoteFeaturesSchema {
+  appHome?: ManifestAppHomeSchema;
+  botUser?: Omit<ManifestBotUserSchema, "display_name">;
+  shortcuts?: ManifestShortcutsSchema;
+  slashCommands?: ManifestSlashCommandsSchema;
+  unfurlDomains?: ManifestUnfurlDomainsSchema;
+  workflowSteps?: ManifestWorkflowStepsSchema;
+}
+
+/** SlackManifestFromType is a type function which takes a parameterT.
+ * It looks at the SlackManifestType which is a discriminated union.
+ * The function returns the type whose discriminant property (slackHosted) matches T
+ */
+export type SlackManifestFromType<T extends boolean> = Extract<
```

Comment:
This isn't used anywhere from what I can see. But it could be useful for devs

Reply:
Maybe we may need to use this function when we trying to change the camel case file in the future, to keep the `SlackManifestType` act according to the discriminant `slackHosted`.